### PR TITLE
Implement Chat Notifications

### PR DESCRIPTION
**Pull request by ReadeP**
## Overview

Implement chat notifications when a user receives a message from their Pear.

## Changes Made
Added functionality to upload FCM token to Pear backend and to deliver a notification to a receiving user in `NetworkManager`

## Test Coverage
Tested on device using Pear dev server

### CODE REVIEW

File: Pear/AppDelegate.swift [hunk 1]
```diff
@@ -51,6 +53,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         // If any sessions were discarded while the application was not running, this will be called shortly after application:didFinishLaunchingWithOptions.
         // Use this method to release any resources that were specific to the discarded scenes, as they will not return.
     }
+
```
Comment (danielVebman) [inline Pear/AppDelegate.swift:56]: delete

File: Pear/Controller/HomeViewController.swift [hunk 2]
```diff
@@ -280,18 +281,20 @@ extension HomeViewController: UICollectionViewDelegateFlowLayout {
...
 //                self.scheduleNotifications(center, day: 2, hour: 8, title: "Meet your new pear!", body: "Set up this week's chat today 😊", aboutPear: true)
 //                self.scheduleNotifications(center, day: 4, hour: 14, title: "Did you reach out yet?", body: "Choose a meeting time with your Pear before it's too late!", aboutPear: true)
 //                self.scheduleNotifications(center, day: 6, hour: 12, title: "How's it going?", body: "New pairings will come out next week! ⌚️", aboutPear: true)
 //                self.scheduleNotifications(center, day: 0, hour: 0, title: "Are you running the latest version of Pear?", body: "Open TestFlight to check for new updates", aboutPear: false)
             }
         }
+        
+        UIApplication.shared.registerForRemoteNotifications()
+        
+
```
Comment (danielVebman) [inline Pear/Controller/HomeViewController.swift:297]: delete?

File: Pear/Controller/HomeViewController.swift [hunk 3]
```diff
@@ -307,6 +310,20 @@ extension HomeViewController: UNUserNotificationCenterDelegate {
...
+    
+    private func getToken() {
+        Messaging.messaging().token { token, error in
+            if let error = error {
+                print(error.localizedDescription)
+            }
+            
+            guard let token = token else {
+                return
+            }
```
Comment (danielVebman) [inline Pear/Controller/HomeViewController.swift:322]: inline

File: Pear/Controller/HomeViewController.swift [hunk 4]
```diff
@@ -280,18 +281,20 @@ extension HomeViewController: UICollectionViewDelegateFlowLayout {
...
+        center.requestAuthorization(options: [.alert, .sound, .badge]) { (granted, error) in
             if granted {
                 center.delegate = self
                 // get rid of previously scheduled notifications
-                center.removeAllDeliveredNotifications()
-                center.removeAllPendingNotificationRequests()
 //                self.scheduleNotifications(center, day: 2, hour: 8, title: "Meet your new pear!", body: "Set up this week's chat today 😊", aboutPear: true)
 //                self.scheduleNotifications(center, day: 4, hour: 14, title: "Did you reach out yet?", body: "Choose a meeting time with your Pear before it's too late!", aboutPear: true)
 //                self.scheduleNotifications(center, day: 6, hour: 12, title: "How's it going?", body: "New pairings will come out next week! ⌚️", aboutPear: true)
 //                self.scheduleNotifications(center, day: 0, hour: 0, title: "Are you running the latest version of Pear?", body: "Open TestFlight to check for new updates", aboutPear: false)
```
Comment (danielVebman) [inline Pear/Controller/HomeViewController.swift:291]: Delete all these comments?

File: Podfile [hunk 5]
```diff
@@ -25,6 +25,7 @@ use_frameworks!
     pod 'SideMenu'
 
     # Messaging
+    pod 'Firebase/Messaging'
```
Comment (danielVebman) [inline Podfile:28]: A-Z why not

File: Pear/Networking/NetworkManager.swift [hunk 6]
```diff
@@ -323,6 +323,50 @@ class NetworkManager {
             }
         }
     }
+    
+    static func updateFCMToken(token: String) {
+
```
Comment (danielVebman) [inline Pear/Networking/NetworkManager.swift:328]: delete

File: Pear/Networking/NetworkManager.swift [hunk 7]
```diff
@@ -323,6 +323,50 @@ class NetworkManager {
...
+            case .success:
+                break
+            case .failure(let error):
+                print(error.localizedDescription)
+            }
+        }
+    }
+    
+    static func deliverNotification(receivingId: Int, message: String) {
+
```
Comment (danielVebman) [inline Pear/Networking/NetworkManager.swift:350]: delete

**Review by danielVebman — APPROVED** [review]
lgtm

**Review by vianxnguyen — APPROVED** [review]
Looks good!

**Review by emarc0314 — APPROVED** [review]
LGTM🗿🔥

**Review by amyqxhuang — APPROVED** [review]
looks good to me!

**Pull request merged**